collections/ansible_collections/cloudkit/service/roles/publish_templates/tasks/vms.yaml

```
@@ -0,0 +1,45 @@
+# VM Template Publishing
+- name: Get list of existing VM templates
+  ansible.builtin.uri:
+    url: "{{ publish_templates_vm_api_endpoint }}"
+    validate_certs: "{{ publish_templates_validate_certs | default(true) | bool }}"
+    headers:
+      authorization: Bearer {{ cloudkit_fulfillment_service_token }}
+  register: vm_template_check
+
+- name: Create list of existing VM template IDs
+  ansible.builtin.set_fact:
+    vm_template_ids: >-
+      {{
+      (vm_template_check.json.items | default([])) | map(attribute='id') | list
+      }}
+
+- name: Update existing VM template
+  loop: "{{ cloudkit_vm_templates }}"
+  loop_control:
+    label: "{{ template.id }}"
+    loop_var: template
+  when: vm_template_check is success and vm_template_check.json.size != 0
+  ansible.builtin.uri:
+    url: "{{ publish_templates_vm_api_endpoint }}/{{ template.id }}"
+    validate_certs: "{{ publish_templates_validate_certs | default(true) | bool }}"
+    method: PATCH
+    headers:
+      authorization: Bearer {{ cloudkit_fulfillment_service_token }}
+    body_format: json
+    body: "{{ template }}"
+
+- name: Create new VM template
+  loop: "{{ cloudkit_vm_templates }}"
+  loop_control:
+    label: "{{ template.id }}"
+    loop_var: template
+  when: cloudkit_vm_templates | length > 0 | default([])
```

```suggestion
  when: template.id not in template_ids
```
we want to create the VM templates that are not already existing in the API.